daemon/daemon.go

```diff
@@ -1056,7 +1058,10 @@ func writeDistributionProgress(cancelFunc func(), outStream io.Writer, progressC
 
 	for prog := range progressChan {
 		if err := progressOutput.WriteProgress(prog); err != nil && !operationCancelled {
-			logrus.Errorf("error writing progress to client: %v", err)
+			// don't log broken pipe errors as this is the normal case when a client aborts
+			if !isBrokenPipe(err) {
+				logrus.Errorf("error writing progress to client: %v", err)
```

Comment:
I don't understand the "may be". If the client disconnects, the pull gets cancelled.

> On Dec 18, 2015, at 10:00, Brian Goff <notifications@github.com> wrote:
> 
> In daemon/daemon.go:
> 
> > @@ -1056,7 +1058,10 @@ func writeDistributionProgress(cancelFunc func(), outStream io.Writer, progressC
> >  
> >  	for prog := range progressChan {
> >  		if err := progressOutput.WriteProgress(prog); err != nil && !operationCancelled {
> > -			logrus.Errorf("error writing progress to client: %v", err)
> > +			// don't log broken pipe errors as this is the normal case when a client aborts
> > +			if !isBrokenPipe(err) {
> > +				logrus.Errorf("error writing progress to client: %v", err)
> Ok, so maybe a logrus.Info("Client disconnected, pull may be cancelled")
> 
> —
> Reply to this email directly or view it on GitHub.
>